code/datums/components/goliath_plate.dm

```diff
@@ -1,18 +1,30 @@
-//Actual armor modification is handled in code/modules/mining/equipment/goliath_hide.dm, this is just the tracker/examine info
-
 /datum/component/goliath_plate
-	var/amount
+	var/amount = 0
 
-/datum/component/goliath_plate/Initialize()
+/datum/component/goliath_plate/Initialize(obj/item/src,mob/user)
 	if(!isatom(parent))
 		return COMPONENT_INCOMPATIBLE
 
 	RegisterSignal(COMSIG_PARENT_EXAMINE, .proc/examine)
-
-	amount = 0
+	RegisterSignal(COMSIG_PARENT_ATTACKBY, .proc/applyplate)
+	applyplate(src,user) //need to call it once since the component does not start applied to the armors
 
 /datum/component/goliath_plate/proc/examine(mob/user)
 	if(amount)
 		to_chat(user, "It has been strengthened with [amount] goliath plate\s.")
 	else
-		to_chat(user, "It can be strengthened with goliath hide plates.")
\ No newline at end of file
+		to_chat(user, "It can be strengthened with goliath hide plates.")
+
+/datum/component/goliath_plate/proc/applyplate(obj/item/I, mob/user)
```

Comment:
Do not merge: at the moment it does not work with mining hardsuit (but works with its helmet and with explorer suit/explorer hood) and I don't know why.

Reply:
mining hardsuit might be overriding it's attackby and not calling ..() which is needed for the signal to be sent

Reply:
Proper fix is to make it so it calls ..() if it doesn't do anything itself

Reply:
That was it, thanks.